src/Customer.java

```diff
@@ -1,14 +1,43 @@
+import customExceptions.CustomerNotFoundException;
+import customExceptions.InvalidNameException;
+import java.util.ArrayList;
+import java.util.List;
 
 public class Customer extends Person {
     private double budget;
     private boolean isBuying;
     private boolean isRenting;
+    private static int nextId = 1500;
+    private int id;
+    private static List<Customer> customers = new ArrayList<>();
     
-    public Customer(String name, String email, String phone, double budget, boolean isBuying, boolean isRenting) {
+    public Customer(String name, String email, String phone, double budget, boolean isBuying, boolean isRenting) throws InvalidNameException {
         super(name, email, phone);
+        if (name.length() < 1) {
```

Comment:
you can use isEmpty() method here - [link](https://www.w3schools.com/java/ref_string_isempty.asp)